grpclog/component.go

```diff
@@ -0,0 +1,253 @@
+// variable and applies all settings specified in the environment variable.
+// Notably, settings for all components with a given prefix are applies before
+// regular component settings.
+func init() {
+	// Pull environment variable data and put in environmentVars and prefixVars
+	if envVar, ok := os.LookupEnv(envName); ok && len(envVar) > 0 {
+		environmentVars, prefixVars = parseEnvironmentVar(envVar)
+	}
+}
+
+// parseEnvironmentVar parses an environment variable string and pulls the component settings data.
+func parseEnvironmentVar(envVar string) (map[string]*componentData, map[string]*componentData) {
+	envVars := map[string]*componentData{}
+	preVars := map[string]*componentData{}
+	varList := strings.Split(envVar, ",")
+	for _, varPair := range varList {
+		varPairList := strings.Split(varPair, ":")
+		if len(varPairList) != 2 {
+			fmt.Fprintf(os.Stderr, "error: could not parse '%v' value '%v', unrecognized key-value pair '%v'\n", envName, envVar, varPair)
+			continue
+		}
+		if cData, ok := parseVar(varPairList[0], varPairList[1]); ok {
+			if varPrefix, ok := getPrefix(varPairList[0]); ok {
+				preVars[varPrefix] = &cData
+			} else {
+				envVars[varPairList[0]] = &cData
+			}
+		} else {
+			fmt.Fprintf(os.Stderr, "error: could not parse '%v' value '%v', unrecognized value '%v'\n", envName, envVar, varPairList[1])
+		}
+	}
+	return envVars, preVars
+}
+
+// apply applies the parameter componentData to the receiver componentData.
+// Sentinel values in the parameter will not be applied.
+func (c *componentData) apply(applyData *componentData) {
+	if applyData.verbosity != sentinel {
+		c.verbosity = applyData.verbosity
+	}
+	if applyData.level != sentinel {
+		c.level = applyData.level
+	}
+}
+
+// parseVar parses a key-value pair from the environment variable. The resulting
+// componentData will have a sentinel value for the verbosity if it is
+// unspecified in the value. Second return value is false if there was an error
+// parsing.
+func parseVar(key string, value string) (componentData, bool) {
+	result := componentData{key, sentinel, sentinel}
+	value = strings.ToUpper(value)
+	switch {
+	case value == "INFO":
+		result.level = levelInfo
+	case value == "WARNING":
+		result.level = levelWarning
+	case value == "ERROR":
+		result.level = levelError
+	case strings.HasPrefix(value, "INFO_"):
+		result.level = levelInfo
+		vStr := strings.TrimPrefix(value, "INFO_")
+		v, err := strconv.Atoi(vStr)
+		if err != nil {
+			return result, false
+		}
+		result.verbosity = v
+	default:
+		return result, false
+	}
+	return result, true
+}
+
+// getPrefix Gets the prefix if s has a wildcard. If s does not have a wildcard,
+// returns "", false.
+func getPrefix(s string) (string, bool) {
+	if strings.HasSuffix(s, "*") {
+		return strings.TrimSuffix(s, "*"), true
+	}
+	return "", false
+}
+
+// InfoDepth performs an info log of args at depth to the component, conditioned
```

Comment:
All of these methods don't need comments anymore since the type is unexported and it exists to satisfy an interface.

Reply:
Removed comments for LoggerV2/DepthLoggerV2 methods.